.github/workflows/ci.yml

```diff
@@ -30,6 +30,9 @@ jobs:
         with:
           tool: cargo-hack,taplo-cli
 
+      - name: Prereqs
+        run: sudo apt-get install -y libelf-dev
```

Comment:
can we back out the changes to this file? we shouldn't need libelf.

Reply:
Ah, I didn't realize we were pulling in libbpf-sys. I suspect you might encounter ideological resistance to a dependency on libbpf.

I think the most pragmatic solution is to add the test in a later PR so that this one doesn't get bogged down in discussions about dependencies. We can discuss alternative solutions there, rather than here.

Reply:
Worth noting that libelf is licensed under LGPL, which has been a barrier to taking a dependency on libbpf-sys in my organization.

Reply:
I don't see the problem with using libbpf-sys (or any other dependency) in _integration tests_? I mean, libbpf-sys is slow to build to I'd prefer if we used something else (and ideally if we added userspace AF_XDP support to aya :P), but I think slow tests is better than no tests for this.

More generally if anything we need more libbpf in integration tests, since we want to ensure we're 100% compatible with it.

Reply:
@tamird then we agree on pulling this test out of the PR, until I find a way to fix the cross-compilation in CI or a way without it. I am very aware of the ideology against libbpf in this project, but as @alessandrod just mentioned, I think it is acceptable for integration test.

I am on vacation abroad with a crappy laptop, I’ll try to work on this but no promises till mid-august.

@irrelevelephant interesting. I am no expert, but even the LGPL is an issue? And is it contagious enough for the integration test to affect the library itself?

Reply:
> @irrelevelephant interesting. I am no expert, but even the LGPL is an issue? And is it contagious enough for the integration test to affect the library itself?

we're not distributing anything that depends on libbpf-sys, libelf etc. In integration tests we can do whatever we want, they're a separate binary, not an end user artifact they're only used by people who work _on_ aya and we're not distributing them in any way

Reply:
> we're not distributing anything that depends on libbpf-sys, libelf etc. In integration tests we can do whatever we want, they're a separate binary, not an end user artifact they're only used by people who work _on_ aya and we're not distributing them in any way

That’s what I suspected, thanks for the confirmation.

Reply:
@tamird the af-xdp test has been pulled out of the pr. this is now ready.

@alessandrod given the new vm runner that uses an ad-hoc initramfs, I doubt we'll ever get libbpf in there :D